### PR TITLE
chore: update golanci-lint to 1.38.0

### PR DESCRIPTION
**Pull request by Unix4ever**
Fix all discovered issues.
Detected couple bugs, fixed them as well.

Signed-off-by: Artem Chernyshev <artem.0xD2@gmail.com>

### CODE REVIEW

File: internal/app/machined/pkg/controllers/k8s/kubelet_static_pod_controller.go [hunk 1]
```diff
@@ -221,7 +221,7 @@ func (ctrl *KubeletStaticPodController) writePod(ctx context.Context, r controll
 
 	renderedPod, err := yaml.Marshal(staticPod.Spec())
 	if err != nil {
-		return nil
+		return err
```
Comment (Unix4ever) [inline internal/app/machined/pkg/controllers/k8s/kubelet_static_pod_controller.go:224]: The first bug.

File: internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer_tasks.go [hunk 2]
```diff
@@ -144,18 +140,14 @@ func MountBPFFS(seq runtime.Sequence, data interface{}) (runtime.TaskExecutionFu
...
-		return nil
+		return mount.Mount(mountpoints)
 	}, "mountBPFFS"
 }
 
 const (
 	memoryCgroup                  = "memory"
 	memoryUseHierarchy            = "memory.use_hierarchy"
-	memoryUseHierarchyPermissions = os.FileMode(400)
+	memoryUseHierarchyPermissions = os.FileMode(0o400)
```
Comment (Unix4ever) [inline internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer_tasks.go:150]: The second bug.

File: internal/app/machined/pkg/system/system.go [hunk 3]
```diff
@@ -18,7 +18,8 @@ import (
 	"github.com/talos-systems/talos/pkg/conditions"
 )
 
-type singleton struct {
```
Comment (smira) [inline internal/app/machined/pkg/system/system.go:21]: Which linter enforces this change? Keeping this type private was supposed to be the way to enforce single instance of this type

**Review by smira — COMMENTED** [review]
:+1: just one question

File: internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go [hunk 4]
```diff
@@ -778,6 +774,7 @@ func (s *Server) List(req *machine.ListRequest, obj machine.MachineService_ListS
 }
 
 // DiskUsage implements the machine.MachineServer interface.
+// nolint:cyclop
```
Comment (AlekSi) [inline internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go:777]:
```suggestion
//nolint:cyclop
```

File: internal/app/machined/pkg/system/system.go (hunk 3 above)
Comment (Unix4ever) [reply internal/app/machined/pkg/system/system.go:21]: let me check

Comment (Unix4ever) [reply internal/app/machined/pkg/system/system.go:21]: that's golint. I guess it makes sense to set `nolint:golint` there instead.

File: internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go (hunk 4 above)
Comment (Unix4ever) [reply internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go:777]: we also need a linter for that whitespace check :D That's a reflex...

**Review by smira — APPROVED** [review]

**Review by smira — APPROVED** [review]

**Pull request merged**